Code/source/ExpressionUtil.cpp

```diff
@@ -1,11 +1,242 @@
+	expression.erase(remove_if(expression.begin(), expression.end(), isspace), expression.end());
+
+	vector<string> operators{ "+", "-", "*", "/", "%" , "(", ")" };
+
+	//Add a space at every occurence of an operator.
+	for (string op : operators) {
+		size_t pos = 0;
+		while (true) {
+			pos = expression.find(op, pos);
+
+			//Finishing condition
+			if (pos == string::npos) {
+				break;
+			}
+
+			//Don't add if at the start.
+			if (pos != 0 && expression[pos] != '(') {
+				expression.insert(pos, " ");
+				pos++;
+			}
+
+			//Account for the actual operator
+			pos++;
+
+			//Don't add if at the end.
+			if (pos >= expression.size()) {
+				break;
+			}
+
+			if (expression[pos-1] != ')') {
+				expression.insert(pos, " ");
+				pos++;
+			}
+		}
+	}
+}
+
+vector<string> ExpressionUtil::getVariables(string infixExpression) {
+	smatch result;
+	regex_search(infixExpression, result, terms);
+
+	vector<string> variableVector;
+
+	for (size_t i = 0; i < result.size(); i++) {
+		bool isValidName = verifyName(result[i]);
+		if (isValidName) {
+			variableVector.push_back(result[i]);
+		}
+	}
+
+	return variableVector;
+}
+
+vector<string> ExpressionUtil::getConstants(string infixExpression) {
+	smatch result;
+	regex_search(infixExpression, result, terms);
+
+	vector<string> constantVector;
+
+	for (size_t i = 0; i < result.size(); i++) {
+		bool isValidInteger = verifyInteger(result[i]);
+		if (isValidInteger) {
+			constantVector.push_back(result[i]);
+		}
+	}
+
+	return constantVector;
+}
+
+//Check if expression given is a valid Infix expression. Perform this step before running convert to Infix.
+bool ExpressionUtil::verifyInfixExpression(string infixExpression) {
+	if (regex_match(infixExpression, validExpression)) {
+		if (checkParenthesis(infixExpression)) {
+			smatch result;
+			regex_search(infixExpression, result, terms);
+
+			for (size_t i = 0; i < result.size(); i++) {
+				bool isValidName = verifyName(result[i]);
+				bool isValidInteger = verifyInteger(result[i]);
+				if (!isValidName && !isValidInteger) {
+					return false;
+				}
+			}
+			return true;
+		}
+		else {
+			return false;
+		}
+	}
+	else {
+		return false;
+	}
+}
+
+//Checks if there is matching parenthesis.
+bool ExpressionUtil::checkParenthesis(string infixExpression) {
+	stack<char> bracketStack;
+
+	for (size_t i = 0; i < infixExpression.length(); i++) {
+		if (infixExpression.at(i) == '(') {
+			bracketStack.push('(');
+		}
+		else if (infixExpression.at(i) == ')') {
+			char stackTop = bracketStack.top();
+			if (stackTop == NULL) {
+				return false;
+			}
+			else {
+				bracketStack.pop();
+			}
+		}
+	}
+
+	if (bracketStack.empty()) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+// Get priority of the operator. If is not a operator return 0.
+int ExpressionUtil::getPriority(string p)
+{
+	char c = p.c_str()[0];
+	if (c == '-' || c == '+')
+		return 1;
+	else if (c == '*' || c == '/' || c == '%')
+		return 2;
+	return 0;
+}
```

Comment:
Using enum might be an easier way to go about this as you associate a value to each individual enum